src/main/java/gregtech/api/worldgen/shape/EllipsoidGenerator.java

```diff
@@ -27,31 +30,47 @@ public EllipsoidGenerator(int radiusMin, int radiusMax) {
 
     @Override
     public void loadFromConfig(JsonObject object) {
-        int[] data = OreConfigUtils.getIntRange(object.get("radius"));
-        this.radiusMin = data[0];
-        this.radiusMax = data[1];
+        int[] radiusData = OreConfigUtils.getIntRange(object.get("radius"));
+        this.radiusMin = radiusData[0];
+        this.radiusMax = radiusData[1];
+
+        JsonElement heightJson = object.get("height");
+        if (heightJson != null) {
+            int[] heightData = OreConfigUtils.getIntRange(heightJson);
+            if (heightData.length == 2) {
+                this.heightMin = heightData[0];
+                this.heightMax = heightData[1];
+            }
+        }
     }
 
     @Override
     public Vec3i getMaxSize() {
-        return new Vec3i(radiusMax * 2, radiusMax, radiusMax * 2);
+        return new Vec3i(radiusMax * 2, heightMax == 0 ? radiusMax : heightMax * 2, radiusMax * 2);
     }
 
     @Override
     public void generate(Random gridRandom, IBlockGeneratorAccess blockAccess) {
         int a = radiusMin >= radiusMax ? radiusMin : (gridRandom.nextInt(radiusMax - radiusMin) + radiusMin);
-        int b = radiusMin >= radiusMax ? radiusMin / 2 : (gridRandom.nextInt(radiusMax - radiusMin) + radiusMin) / 2;
-        int c = radiusMin >= radiusMax ? radiusMin : (gridRandom.nextInt(radiusMax - radiusMin) + radiusMin);
+        int b;
+        if (heightMin == 0) {
+            // default if no height prop provided (also: originally coded behavior)
+            b = radiusMin >= radiusMax ? radiusMin / 2 : (gridRandom.nextInt(radiusMax - radiusMin) + radiusMin) / 2;
+        } else {
+            b = heightMin >= heightMax ? heightMin : (gridRandom.nextInt(heightMax - heightMin) + heightMin);
+        }
+        int c = a; // somewhat pointless, but establishes pattern for other (non-symmetrical) shapes.  Ellipsoids will always be circular as viewed from above.
```

Comment:
You understand that you're breaking original generator by forcing same size on both axis, right? Originally, it randomly chose radius by X, Y, Z axes separately, forming ellipsoid shape, and you made it sphere by forcing X to equal Z.
Lookup and compare what you got old way and the new one. Ellipsoid doesn't look like a sphere and is not forced to be circular when viewed from above. Only body forced to be circular when viewed from above is sphere.